cumulus/polkadot-parachain/polkadot-parachain-lib/src/nodes/manual_seal.rs

```diff
@@ -0,0 +1,234 @@
+};
+use codec::Encode;
+use cumulus_client_parachain_inherent::{MockValidationDataInherentDataProvider, MockXcmConfig};
+use cumulus_primitives_core::ParaId;
+use sc_consensus::{DefaultImportQueue, LongestChain};
+use sc_consensus_manual_seal::rpc::{ManualSeal, ManualSealApiServer};
+use sc_network::NetworkBackend;
+use sc_service::{build_polkadot_syncing_strategy, Configuration, PartialComponents, TaskManager};
+use sc_telemetry::TelemetryHandle;
+use sp_runtime::traits::Header;
+use sp_timestamp::Timestamp;
+use std::{marker::PhantomData, sync::Arc};
+
+/// Build the import queue for a manual seal node.
+pub struct BuildManualSealImportQueue<Block, RuntimeApi>(PhantomData<(Block, RuntimeApi)>);
+
+impl<
+		Block: NodeBlock,
+		RuntimeApi: ConstructNodeRuntimeApi<Block, ParachainClient<Block, RuntimeApi>>,
+	> BuildImportQueue<Block, RuntimeApi> for BuildManualSealImportQueue<Block, RuntimeApi>
+{
+	fn build_import_queue(
+		client: Arc<ParachainClient<Block, RuntimeApi>>,
+		_block_import: ParachainBlockImport<Block, RuntimeApi>,
+		config: &Configuration,
+		_telemetry_handle: Option<TelemetryHandle>,
+		task_manager: &TaskManager,
+	) -> sc_service::error::Result<DefaultImportQueue<Block>> {
+		Ok(sc_consensus_manual_seal::import_queue(
+			Box::new(client.clone()),
+			&task_manager.spawn_essential_handle(),
+			config.prometheus_registry(),
+		))
+	}
+}
+
+pub struct ManualSealNode<NodeSpec>(PhantomData<NodeSpec>);
+
+impl<NodeSpec: NodeSpecT> BaseNodeSpec for ManualSealNode<NodeSpec> {
+	type Block = NodeSpec::Block;
+	type RuntimeApi = NodeSpec::RuntimeApi;
+	type BuildImportQueue = BuildManualSealImportQueue<Self::Block, Self::RuntimeApi>;
+}
+
+impl<NodeSpec: NodeSpecT> ManualSealNode<NodeSpec> {
+	pub fn new() -> Self {
+		Self(Default::default())
+	}
+
+	pub fn start_node<Net>(
+		&self,
+		config: Configuration,
+		para_id: ParaId,
+		block_time: u64,
+	) -> sc_service::error::Result<TaskManager>
+	where
+		Net: NetworkBackend<NodeSpec::Block, Hash>,
+	{
+		let PartialComponents {
+			client,
+			backend,
+			mut task_manager,
+			import_queue,
+			keystore_container,
+			select_chain: _,
+			transaction_pool,
+			other: (_, mut telemetry, _),
+		} = Self::new_partial(&config)?;
+		let select_chain = LongestChain::new(backend.clone());
+
+		let mut net_config = sc_network::config::FullNetworkConfiguration::<_, _, Net>::new(
+			&config.network,
+			config.prometheus_config.as_ref().map(|cfg| cfg.registry.clone()),
+		);
+		let metrics = Net::register_notification_metrics(
+			config.prometheus_config.as_ref().map(|cfg| &cfg.registry),
+		);
+
+		let syncing_strategy = build_polkadot_syncing_strategy(
+			config.protocol_id(),
+			config.chain_spec.fork_id(),
+			&mut net_config,
+			None,
+			client.clone(),
+			&task_manager.spawn_handle(),
+			config.prometheus_config.as_ref().map(|config| &config.registry),
+		)?;
+
+		let (network, system_rpc_tx, tx_handler_controller, start_network, sync_service) =
```

Comment:
We need to make sure here that network does not try to sync. Maybe setting the in/out peers to 0 or maybe we have a mock somewhere. The dev node does not really need networking.

Problems occured when I was playing with this and invoked for example `polkadot-parachain --dev-block-time 3000 --tmp --chain asset-hub-rococo.json`. The node found some peers and tried to import blocks from them, which failed.

Reply:
Good point ! Done.